src/containers/sound-editor.jsx

```diff
@@ -0,0 +1,110 @@
+const bindAll = require('lodash.bindall');
+const PropTypes = require('prop-types');
+const React = require('react');
+
+const {connect} = require('react-redux');
+
+const {computeRMS} = require('../lib/audio/audio-util.js');
+
+const SoundEditorComponent = require('../components/sound-editor/sound-editor.jsx');
+const AudioBufferPlayer = require('../lib/audio/audio-buffer-player.js');
+
+const getChunkLevels = (samples, chunkSize = 1024) => {
+    const sampleCount = samples.length;
+    const chunkLevels = [];
+    for (let i = 0; i < sampleCount; i += chunkSize) {
+        const maxIndex = Math.min(sampleCount - 1, i + chunkSize);
+        chunkLevels.push(computeRMS(samples.slice(i, maxIndex), true));
+    }
+    return chunkLevels;
+};
+
+class SoundEditor extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleStoppedPlaying',
+            'handleChangeName',
+            'handlePlay',
+            'handleStopPlaying',
+            'handleUpdatePlayhead'
+        ]);
+        this.state = {
+            playhead: null, // null is not playing, [0 -> 1] is playing percent
+            chunkLevels: getChunkLevels(this.props.samples)
+        };
+    }
+    componentDidMount () {
+        this.audioBufferPlayer = new AudioBufferPlayer(this.props.samples, this.props.sampleRate);
+    }
+    componentWillReceiveProps (newProps) {
+        if (newProps.soundIndex !== this.props.soundIndex) {
+            this.audioBufferPlayer.stop();
```

Comment:
It doesn't, but the next line reassigns so the whole thing will get garbage collected.